esphome/components/sen21231/sensor.py

```diff
@@ -0,0 +1,31 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import i2c, sensor
+from esphome.const import ICON_MOTION_SENSOR, UNIT_EMPTY
+
+CODEOWNERS = ["@shreyaskarnik"]
+DEPENDENCIES = ["i2c"]
+
+sen21231_sensor_ns = cg.esphome_ns.namespace("sen21231_sensor")
+Sen21231Sensor = sen21231_sensor_ns.class_(
+    "Sen21231Sensor", cg.PollingComponent, i2c.I2CDevice
+)
+
+CONFIG_SCHEMA = (
+    sensor.sensor_schema(
+        unit_of_measurement=UNIT_EMPTY, icon=ICON_MOTION_SENSOR, accuracy_decimals=1
+    )
+    .extend(
+        {
+            cv.GenerateID(): cv.declare_id(Sen21231Sensor),
+        }
+    )
+    .extend(cv.polling_component_schema("30s"))
```

Comment:
We always try to default sensors to 60s unless there is a specific reason? Its easy for a user to override this in yaml.
```suggestion
    .extend(cv.polling_component_schema("60s"))
```